lib/refresh.js

```diff
@@ -1,142 +1,141 @@
 'use strict';
 
-const AuthTokenRefresh = {};
```

Comment:
This looks big only because I had to put the code inside a constructor function. There are no functional changes.

Reply:
I wonder if, instead of creating instance variables on `this`, we should prefer attaching to the `prototype` instead. So, we would have (for example):

```
AuthTokenRefresh.prototype.use = function (name, strategy, options) {
  // etc
}
```

As a result, the code changes in this file will be much more minimal, as it will boil down to:

- Changing `const AuthTokenRefresh = {};` to be a function instead
- Setting `this.strategies = {}` in the constructor 
- Using prototype methods instead of properties on an object
- Using `this.strategies` inside the prototype methods instead of `_strategies`

Would this work? What do you think?